zfgrep: add page (#8708)
tldr-pages/tldr: pages/common/zfgrep.md

# zfgrep

> Matches fixed strings in possibly compressed files.

> > Equivalent to `grep -F` but first decompresses input if necessary.

Reviewer: ```suggestion
> Equivalent to `grep -F` with input decompressed first if necessary.
```
I think this would be better

> > More information: <https://www.gnu.org/software/grep/manual/grep.html>.

Reviewer 2: I think we should point to `zfgrep` instead of `grep` :)
```suggestion
> More information: <https://manned.org/zfgrep>.
```

- Search for an exact string in a file:

> `zfgrep {{search_string}} {{path/to/file}}`

Reviewer 2: In the `grep` page, we use `search_pattern`, i think we should be consistent :)
```suggestion
`zfgrep {{search_pattern}} {{path/to/file}}`
```

Author: Mmm, I could make that change if you really wanted it, but really the whole point of the "F" variant of the grep commands is that the thing you are searching for is taken as a literal string, NOT as a pattern.  So I would prefer to leave it as search_string if up to me...  Thoughts?

Reviewer: @navarroaxel Should we use `search_string` or `search_pattern` ?

- Count the number of lines that match the given string in a file:

> `zfgrep --count {{search_string}} {{path/to/file}}`

Reviewer 2: ```suggestion
`zfgrep --count {{search_pattern}} {{path/to/file}}`
```

Author: Same literal string vs pattern comment as above.

- Show the line number in the file along with the matching lines:

> `zfgrep --line-number {{search_string}} {{path/to/file}}`

Reviewer 2: ```suggestion
`zfgrep --line-number {{search_pattern}} {{path/to/file}}`
```

Author: Same literal string vs pattern comment as above.

- Display all lines except those that contain the search string:

> `zfgrep --invert-match {{search_string}} {{path/to/file}}`

Reviewer 2: ```suggestion
`zfgrep --invert-match {{search_pattern}} {{path/to/file}}`
```

Author: Same literal string vs pattern comment as above.

- List only filenames whose content matches the search string at least once:

> `zfgrep --files-with-matches {{search_string}} {{path/to/file1 path/to/file2 ...}}`

Reviewer 2: ```suggestion
`zfgrep --files-with-matches {{search_pattern}} {{path/to/file1 path/to/file2 ...}}`
```

Author: Same literal string vs pattern comment as above.
